datafusion/functions/src/regex/regexpreplace.rs

```diff
@@ -94,14 +98,30 @@ impl Default for RegexpReplaceFunc {
 
 impl RegexpReplaceFunc {
     pub fn new() -> Self {
-        use DataType::*;
+        use TypeSignature::*;
+        use TypeSignatureClass::*;
         Self {
             signature: Signature::one_of(
                 vec![
-                    TypeSignature::Exact(vec![Utf8, Utf8, Utf8]),
```

Comment:
I personally recommend supporting only "all UTF8" , "all largeUtf8" and "all Utf8View"

Supporting all combinations I think is likely to generate lots of code that will make the binary bigger but not be used by anyone